process_hierarchy.py

```diff
@@ -0,0 +1,478 @@
+    hierarchy.json    - 3 layers institution, field-of-science, pi
+    groups.json       - information linking project ids to the pi 
+    pi2project.json   - this links cloud-projects ids to projects ids
+    names.json        - links ids to project names
+"""
+# pylint: disable=line-too-long invalid-name
+import json
+import pprint
+import copy
+import hashlib
+import moc_db_helper_functions
+
+
+def get_data_from_coldfront():
+    """
+    This links PI with project/cloud_project.
+
+    This fetches the allocation data from coldfront with the format of:
+    [
+    {
+        "id": <num>,                      # coldfront allocation id
+        "project": {
+            "id": <num>,                  # coldfront project id
+            "title": <str>,               # colfront project title
+            "pi": "willjt@bu.edu",        # principle investigator
+            "description": <str>,         # coldfront project description
+            "field_of_science": <str>,    # coldfront project field of science
+            "status": "New"               # coldfront project status, ("New", ...)
+        },
+        "description": "",                # coldfront allocation
+        "resource": {
+            "name": "NERC-OCP",           # coldfront allocation resource
+            "resource_type": "OpenShift"  # coldfront allocation resource type
+        },
+        "status": "Active",               # coldfront allocation status
+        "attributes": {
+            "Allocated Project ID": str,  # rosource specific project id
+            "Allocated Project Name": str # resource specific project name
+        }
+    },
+    ...
+    ]
+
+    """
+    with open("../coldfront_data.json", "r", encoding="utf-8") as coldfront_file:
+        data = json.load(coldfront_file)
+        return data
+    return None
+
+
+def get_data():
+    """this just fetches idealized data and will eventually go away!"""
+    with open("../hierarchy_data.json", "r", encoding="utf-8") as hierarchy_file:
+        data = json.load(hierarchy_file)
+        for rec in data:
+            rec["status"] = "Active"
+        return data
+    return None
+
+
+def create_hierarchy_db(cursor):
+    """creates the tables of the hierarchy database"""
+    cursor.execute("create database hierarchy_db", None)
+    cursor.execute(
+        "create table hierarchy_db.hierarchy_rec ( \
+            id varchar(256), \
+            create_ts timestamp, \
+            type varchar(100) not null, \
+            name varchar(500) not null, \
+            status varchar(100), \
+            display_name varchar(500), \
+            parent_id varchar(256), \
+            primary key (id, create_ts) \
+        )",
+        None,
+    )
+    cursor.execute(
+        "create sequence hierarchy_db.hierarchy_db_id_seq start with 3 increment by 1;"
+    )
+    # The sha256 field is a compsite key based on all
+    cursor.execute(
+        "create table hierarchy_db.hierarchy_keys ( \
```

Comment:
I don't see you using this table anywhere. Is it necessary?

Reply:
I created that table to hold the ids that are constructed using the digested/hashed ids.  However I am currently assigning IDs using a sequencer, so I may just use the ids that I assign as opposed to using the digest.

This will be deleted if I don't need it.